streams/src/main/java/org/apache/kafka/streams/processor/StateStore.java

```diff
@@ -61,7 +84,9 @@
      * @throws IllegalStateException If store gets registered after initialized is already finished
      * @throws StreamsException if the store's change log does not contain the partition
      */
-    void init(ProcessorContext context, StateStore root);
+    default void init(final StateStoreContext context, final StateStore root) {
+        init(StoreToProcessorContextAdapter.adapt(context), root);
+    }
```

Comment:
Proposing to add this new version of `init` since the old `ProcessorContext` is slated to be deprecated once KIP-478 is fully implemented.

Rather than just accepting the whole new ProcessorContext, I'm proposing to add a new StateStoreContext to avoid leaking a bunch of methods that should really only be invoked in Processor implementations.